supervisor/update.go

```diff
@@ -44,11 +46,16 @@ func (h *UpdateTask) Handle(e *Task) error {
 }
 
 type UpdateProcessTask struct {
-	s *Supervisor
+	baseTask
+	ID         string
+	PID        string
+	CloseStdin bool
+	Width      int
+	Height     int
 }
 
-func (h *UpdateProcessTask) Handle(e *Task) error {
-	i, ok := h.s.containers[e.ID]
+func (s *Supervisor) updateProcess(t *UpdateProcessTask) error {
+	i, ok := s.containers[t.ID]
```

Comment:
No since it is inside the event loop it is safe from concurrent access.  I think i still need to document this and what guarantees you get when accessing this from inside the main loop vs in concurrent workers or other.